### ISSUE TITLE
No need for "typedef" for array methods

### ISSUE THREAD

**Issue by daletune** [palantir/tslint#670]
With options typedef ("parameter")
In case:
```typescript
var myArray: Array<number> = [1, 3, 4, 5, 6];

myArray.forEach(item => {
//...
});
```
We have warning "expected parameter 'item' to have a typedef" but myArray collection items already aware of  type. This scenario also applied to other array methods (some, every, find, reduce, etc.)

Is there any good solution or maybe I miss something?
Thank you.

**Comment by adidahiya** [comment]
The typedef rule doesn't do any type inspection; its main motivation in general is to aid code readability when scanning over a file. This looks like a duplicate of #333.
